### DISCUSSION TITLE
In eBPF how to get tracepoint data which type is __data_loc

### DISCUSSION THREAD

**Discussion by liuzikai0709** [bpftrace/bpftrace#2531] [Q&A]
`format:
	field:unsigned short common_type;	offset:0;	size:2;	signed:0;
	field:unsigned char common_flags;	offset:2;	size:1;	signed:0;
	field:unsigned char common_preempt_count;	offset:3;	size:1;	signed:0;
	field:int common_pid;	offset:4;	size:4;	signed:1;

	field:__data_loc char[] name;	offset:8;	size:4;	signed:0;
	field:unsigned long mbps;	offset:16;	size:8;	signed:0;
	field:unsigned long freq;	offset:24;	size:8;	signed:0;
	field:unsigned long up_thres;	offset:32;	size:8;	signed:0;
	field:unsigned long down_thres;	offset:40;	size:8;	signed:0;

print fmt: "dev: %s, mbps = %lu, freq = %lu, up = %lu, down = %lu", __get_str(name), REC->mbps, REC->freq, REC->up_thres, REC->down_thres`

I want to get tracepoint data in __data_loc,  but  I've tried many methods, but I haven't gotten it。Thank you for your help

**Comment by dxuuu** [comment]
Hi, support was added for `__data_loc` fields in https://github.com/iovisor/bpftrace/pull/1542 . I think `args->name` should just work, no? If not, I can take a closer look.
